bioptim/limits/objective_functions.py

```diff
@@ -174,15 +174,17 @@ class Functions:
             pass
 
         @staticmethod
-        def add_to_penalty(ocp, _, val, penalty, **extra_param):
+        def add_to_penalty(ocp, _, val, penalty, dt=0, target=None, **extra_param):
             """
-            Adds an objective.
-            :param val: Value to be optimized. (MX.sym from CasADi)
+            Adds objective J to objective array nlp.J[penalty] or ocp.J[penalty] at index penalty.
+            :param J: Objective. (dict of [val, target, weight, is_quadratic])
             :param penalty: Index of the objective. (integer)
-            :param weight: Weight of the objective. (float)
-            :param quadratic: If True, value is squared (bool)
             """
-            ObjectiveFunction.add_to_penalty(ocp, None, val, penalty, dt=1)
+            val = val
+            target = target if target is not None else None
+            J = {"objective": penalty, "val": val, "target": target, "dt": dt}
+
+            ocp.J[penalty.idx].append(J)
```

Comment:
Why have you removed the call to add_to_penalty, just to copy what it is doing?

Reply:
I suggest to revert these target stuff as input parameter, since it is already dealt with extra_param...

Reply:
solved in 808d4b4bc43f2fa8d0b7d76681ea1ffac0ea8898